### PR TITLE
[HOTFIX] carbon reader support open another reader without closing previous reader

### PR DESCRIPTION
**Pull request by ajantha-bhat**
[HOTFIX] carbon reader support open another reader without closing previous reader

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed? No
 
 - [ ] Any backward compatibility impacted? No
 
 - [ ] Document update required? NA

 - [ ] Testing done
updated UT       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. NA

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/8901/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/836/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/641/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/644/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/8907/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/839/

File: store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReaderBuilder.java [hunk 1]
```diff
@@ -115,11 +115,13 @@ public CarbonReaderBuilder withHadoopConf(Configuration conf) {
...
     }
     CarbonTable table;
-    if (filterExpression != null) {
-      table = CarbonTable.buildTable(tablePath, tableName, hadoopConf);
-    } else {
-      table = CarbonTable.buildDummyTable(tablePath);
-    }
+    // now always infer schema. TODO:Refactor in next version.
+    table = CarbonTable.buildTable(tablePath, tableName, hadoopConf);
+    //    if (filterExpression != null) {
```
Comment (xuchuanyin) [inline store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReaderBuilder.java]: If it is useless, just remove them. We are using git, no need to comment the useless code

Comment (ajantha-bhat) [reply store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReaderBuilder.java]: yeah. OK

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/843/

**Comment by ravipesala** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/866/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.1, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/8934/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/675/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**